src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -2475,22 +2510,58 @@ public async Task<bool> ApplyUpdates(int moduleId, string dmeta, string dil, str
             return true;
         }
 
-        public async Task<byte[][]> GetBytesFromAssemblyAndPdb(string assemblyName, CancellationToken token)
+        public async Task<bool> HasDebugInfoLoadedByRuntime(string assemblyName, CancellationToken token)
+        {
+            var assemblyId = await GetAssemblyId(assemblyName, token);
+            using var commandParamsWriter1 = new MonoBinaryWriter();
+            commandParamsWriter1.Write(assemblyId);
+            using var retDebuggerCmdReader1 = await SendDebuggerAgentCommand(CmdAssembly.HasDebugInfoLoaded, commandParamsWriter1, token);
+            return retDebuggerCmdReader1.ReadByte() == 1;
+        }
+
+        public async Task<AssemblyAndPdbData> GetDataFromAssemblyAndPdb(string assemblyName, bool ignoreJMC, CancellationToken token)
         {
+            if (!assemblyName.StartsWith("System.Private.CoreLib") && !ignoreJMC && proxy.JustMyCode && !(await HasDebugInfoLoadedByRuntime(assemblyName, token)))
```

Comment:
Also, I would suggest maybe splitting this only multiple lines, and comments for why the different parts of the condition are needed.